src/Microsoft.Azure.AppConfiguration.Functions.Worker/AzureAppConfigurationRefreshExtensions.cs

```diff
@@ -23,14 +23,22 @@ public static class AzureAppConfigurationRefreshExtensions
         /// <param name="builder">An instance of <see cref="IFunctionsWorkerApplicationBuilder"/></param>
         public static IFunctionsWorkerApplicationBuilder UseAzureAppConfiguration(this IFunctionsWorkerApplicationBuilder builder)
         {
+            IServiceProvider serviceProvider = builder.Services.BuildServiceProvider();
+            IConfigurationRefresherProvider refresherProvider = serviceProvider.GetService<IConfigurationRefresherProvider>();
+
             // Verify if AddAzureAppConfiguration was done before calling UseAzureAppConfiguration.
             // We use the IConfigurationRefresherProvider to make sure if the required services were added.
-            if (!builder.Services.Any(service => service.ServiceType == typeof(IConfigurationRefresherProvider)))
+            if (refresherProvider == null)
+            {
+                throw new InvalidOperationException($"Unable to find the required services. Please add all the required services by calling '{nameof(IServiceCollection)}.{nameof(AzureAppConfigurationExtensions.AddAzureAppConfiguration)}()' in the application startup code.");
```

Comment:
> '{nameof(IServiceCollection)}.{nameof(AzureAppConfigurationExtensions.AddAzureAppConfiguration)}()'

Any reason we do it differently in this message compared to the one used in ASP.NET middleware?

Reply:
I tried to change it, but then some using errors came up that were preventing me from using the second `nameof()`. I added the same using statements as the functions worker file and it wouldn't recognize it still.

Reply:
Unfortunately, the name of this class in AspNetCore package is same as the name of Extensions class with AddAzureAppConfiguration method. This class in AspNetCore package should've been called `AzureAppConfigurationRefreshExtensions`, but it would be a breaking change to modify the name now. 
Try using full class name along with namespace to explicitly point to the right class:
`nameof(Microsoft.Extensions.Configuration.AzureAppConfigurationExtensions.AddAzureAppConfiguration)`

Reply:
Looks like it works, no error and it gets the method name